From 4be7fe2e5f533295a17a8ee395a7f888c8300aa2 Mon Sep 17 00:00:00 2001
From: "siva.kannappan" <siva.kannappan@lge.com>
Date: Fri, 14 Aug 2020 11:15:05 +0000
Subject: [PATCH] Fix single instance crash on web application

:Release Notes:
Fix single instance crash on web application

:Detailed Notes:
Umediaserver receiving app_id as app_id + display_id. So on LSRegister
call it's failed for the same. Now removed display_id from app_id and
passed to LSRegister. Will modify this logic to support multi-instance
as part of this epic PLAT-115279.

:Testing Performed:
QWA-2609, QWA-1171, QWA-562, QWA-791,QWA-792, QWA-793,
QWA-429, QWA-434, QWA-437, QWA-2408

:QA Notes:
None.

:Issues Addressed:
[PLAT-116454] [webOS OSE]: When clicked play button in secondary screen,
              The Primary and secondary screen is crashed.
[PLAT-116807] [webOS OSE]: After side loading the link, primary screen
              is crashed.
[PLAT-116864] [webOS OSE]: Web browser is crashed, while playing HTML5
              video

Change-Id: If0a9156f5305dc0a6d42bf60a034e0807cda98c1
---
 src/media_client/uMediaClient.cpp | 11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

diff --git a/src/media_client/uMediaClient.cpp b/src/media_client/uMediaClient.cpp
index 2955387..8c79c41 100644
--- a/src/media_client/uMediaClient.cpp
+++ b/src/media_client/uMediaClient.cpp
@@ -64,13 +64,20 @@ uMediaClient::uMediaClient(bool rawEvents, UMSConnectorBusType bus, const std::s
 {
 	//prefix of uid should be '-' when the client is created under app privileged case
 	std::string uid = GenerateUniqueID()();
-	if(!m_app_connection_id.empty())
+	if(!m_app_connection_id.empty()){
 		std::replace_if(uid.begin(), uid.begin() + 1, [](char c) { return c == '_'; }, '-');
+		/*Umediaserver receiving app_id as app_id + display_id. So on LSRegister
+		call it's failed for the same. So removing display_id from app_id*/
+		if(isdigit(m_app_connection_id.back())) {
+			m_app_connection_id = m_app_connection_id.substr(0,m_app_connection_id.length()-1);
+			}
+		}
 
 	_log.setUniqueId(uid);
 	std::string process_connection_id =
 		(!m_app_connection_id.empty() ? m_app_connection_id : MEDIA_CLIENT_CONNECTION_BASE_ID) + uid;
-	LOG_INFO(_log, "connection-id", "create ums client with connection Id : %s", process_connection_id.c_str());
+	LOG_INFO(_log, "connection-id", "create ums client with connection Id : %s\t app_id : %s",
+		process_connection_id.c_str(), m_app_connection_id.c_str());
 
 	context = g_main_context_new();
 	gmain_loop = g_main_loop_new(context, false);
-- 
2.17.1

